---
title: "Bioconductor Methylation Workflow"
author:
- J. Jedediah Smith
- Gerard Duncan
- Ian Lacy
format: revealjs
editor: visual
---

## 1.1 Background

-   Epigenetics: The study of how your behaviors and environment can change the way your genes work.

-   Methylation: Methyl group gets added to DNA, at C/G bases. Associated with development and disease.

-   Common Methylation Research

    -   Machine learning algorithm to classify tumors.

    -   Physical activity and epigenetic markers.

## 1.2 Workflow Introduction

-  We used the Bioconductor Methylation Array Analysis workflow by Jovana Maksimovic. All examples and code are attributed to them.

-  CpG dinucleotides are DNA sites where a C nucleotide is followed by a G one. This is where methylation occurs.

-  Two CpG measurements determine methylation proportions: methylated intensity (M) and unmethylated intensity (U).

    -   `GetBeta` function in `minfi` to get these values 

## 2.1 Obtaining the Data

-  Example data used here is bundled with the package and easy to access. You do not need to install anything extra.

-  Alternatively, the data can be acquired with `figshare`.

## 2.2 Load the Data

-  There are many methlation specific packages used in this workflow, including `minifi`, `minfiData`, and `dmrcate`.

-  Annotation and experiment data is loaded in with the `read.metharry.sheet` function.

## 2.3 Quality Control

-  Helps to ensure all data is high quality.

-  Calculate detection p-value with the `detP` function.

-  Plot the mean using bar graphs.

-  Generate a report using the `QcReport` function from `minfi`.

## Figure 2.3

![Quality control bar graph of data.](./BioshockDoc_files/figure-html/unnamed-chunk-9-1.png)

## 2.4 Normalization

-  Helps deal with variability between samples.

-  Many different methods even within just `minfi` package.

-  The `preprocessFunnorm` function can be used to detect global methylation differences.

-  The `preprocessQuantile` function is more suited to samples with no global differences.

-  Visualize the before and after data with density plots.

## Figure 2.4

![Density plots comparing raw and normalized data.](./BioshockDoc_files/figure-html/unnamed-chunk-16-1.png)

## 2.5 Data Exploration

-  Visualization of methlation data is often done with multi-dimensional scaling plots (MDS). They look at the similarities and differences between the various samples.
    
    -   MDS is based on principal components analysis (PCA), a data reduction method that makes it easier to interpret large datasets.
    
-  The workflow utilizes the `plotMDS` function to plot data.

## Figure 2.5a

![MDS plots to visualize relationships between samples.](./BioshockDoc_files/figure-html/unnamed-chunk-17-1.png)

## Figure 2.5b

![MDS plots to expose sources of variation in data.](./BioshockDoc_files/figure-html/unnamed-chunk-18-1.png)

## 2.6 Filtering

-  Filter out DNA probes that interfere with analysis.

-  Begin by ordering the probes, then removing any probes that have failed in one or more samples.

-  While this dataset is entirely composed of male donors, it is possible to filter both X and Y chromosomes.

-  Single nucleotide polymorphisms (SNPs) can also be filtered out of probes using `minifi` functions.

-  After filtering, re-examine the data using MDS plots to see if the relationship between samples changed.

-  Also calculate and plot the M and Beta values.

## Figure 2.6a
![MDS plots of filtered data.](./BioshockDoc_files/figure-html/unnamed-chunk-24-1.png)

## Figure 2.6b
![Beta Value and M-Value density plots.](./BioshockDoc_files/figure-html/unnamed-chunk-26-1.png)

## 2.7 Probe-Wise Differential Methylation Analysis

-  The focus of this dataset is discovering differentially methylated probes between four different cell types.

-  In this workflow, we analyzed the M-values in limma. This allowed us to obtain moderated t-statistics and associated p-values for each CpG site.

-  Based on a false discovery rate of 5%, we found 3000 significantly differentially methylated CpGs in the naïve vs rTreg comparison. The rTreg/ act_rTreg did not show any significant differential methylation.

## Figure 2.7
![Table of probe-wise analysis data.](./tableforian.png)

## 2.8 DMR Analysis

-   Probe-wise differential methylation analysis is useful but not sufficent to analyze an entire region.
-   The `dmrcate` package helps us identify and explore differential methylated regions (DMRs).
    -   Other functions like `dmrFind` and `bumphunter` are similar but use statistical methods and run slower.
-   Generated figure maps methlated regions against the chromosome length and known genes.
    -   Need annotated chromosome data.

## Figure 2.8
![The `dmrcate` figure identifying our DMR against CpG prodes, gene loction, and chromosome.](./BioshockDoc_files/figure-html/unnamed-chunk-34-1.png)

## 2.9 Customizing Visualizations

-   The `Gviz` package allows for more precise customization of data visualization plots.
    -   Its package vignette is very extensive with a with a variety of different functions.
-   An important feature utilized by this workflow is to zoom in on an identified DMR.
    -   Need genomic coordinates from `dmrcate`.
    -   Need annotated chromosome data.
-   Allows for clearer presentation of results.

## Figure 2.9
![The `Gviz` figure giving us a close-up of our DMR and CpGs.](./BioshockDoc_files/figure-html/unnamed-chunk-44-1.png)

## 3.1 Conclusions

**Things Learned**

-   Methylation is complicated
-   Bioconductor has a vast number of packages and workflows
-   There are many tools available to visualize data

**Challenges**

-   Undoing commits with GitKraken
-   Updating R and RStudio
-   Getting the document to render
