<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Bioshock Doc</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="BioshockDoc_files/libs/clipboard/clipboard.min.js"></script>
<script src="BioshockDoc_files/libs/quarto-html/quarto.js"></script>
<script src="BioshockDoc_files/libs/quarto-html/popper.min.js"></script>
<script src="BioshockDoc_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="BioshockDoc_files/libs/quarto-html/anchor.min.js"></script>
<link href="BioshockDoc_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="BioshockDoc_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="BioshockDoc_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="BioshockDoc_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="BioshockDoc_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Bioshock Doc</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>J. Jedediah Smith </p>
             <p>Gerard Duncan </p>
             <p>Ian Lacy </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">1.1 Introduction</h2>
<p><strong>Epigenetics and Methylation</strong></p>
<ul>
<li><p>Epigenetics: The study of how your behaviors and environment can cause changes that affect the way your genes work.</p></li>
<li><p>Methylation: Methyl group gets added to DNA, at Cytosine/Guanine bases. Associated with development and disease.</p></li>
</ul>
<p><strong>Common Methylation Research</strong></p>
<ul>
<li><p>Dr.&nbsp;Phillip Jurmeister: Machine learning algorithm to classify sinonasal tumors</p></li>
<li><p>Dr.&nbsp;Glen E Duncan: Physical activity and epigenetic markers</p></li>
</ul>
<p><strong>Methylation Level Measurements</strong></p>
<ul>
<li><p>Two measurements for each CpG: methylated intensity(M) and unmethylated intensity(U) for determining methylation proportions&nbsp;</p></li>
<li><p><code>GetBeta</code> function in <code>minfi</code> to get these values&nbsp;</p></li>
</ul>
</section>
<section id="loading-the-workflow-package" class="level2">
<h2 class="anchored" data-anchor-id="loading-the-workflow-package">1.2 Loading the Workflow Package</h2>
<p>The is the initial step we do to get the methlation array package installed. You can read more about this workflow on the bioconductor website.</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/workflows/html/methylationArrayAnalysis.html" class="uri">https://bioconductor.org/packages/release/workflows/html/methylationArrayAnalysis.html</a></li>
</ul>
<div class="cell">

</div>
</section>
<section id="obtain-the-data" class="level2">
<h2 class="anchored" data-anchor-id="obtain-the-data">2.1 Obtain the Data</h2>
<p>This is how we obtained the example data. It is part of the package so you do not need to install anything extra.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up a path to the data directory</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>dataDirectory <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">"extdata"</span>, <span class="at">package =</span> <span class="st">"methylationArrayAnalysis"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># List the files</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(dataDirectory, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "48639-non-specific-probes-Illumina450k.csv"
 [2] "5975827018/5975827018_R06C02_Grn.idat"     
 [3] "5975827018/5975827018_R06C02_Red.idat"     
 [4] "6264509100/6264509100_R01C01_Grn.idat"     
 [5] "6264509100/6264509100_R01C01_Red.idat"     
 [6] "6264509100/6264509100_R01C02_Grn.idat"     
 [7] "6264509100/6264509100_R01C02_Red.idat"     
 [8] "6264509100/6264509100_R02C01_Grn.idat"     
 [9] "6264509100/6264509100_R02C01_Red.idat"     
[10] "6264509100/6264509100_R02C02_Grn.idat"     
[11] "6264509100/6264509100_R02C02_Red.idat"     
[12] "6264509100/6264509100_R03C01_Grn.idat"     
[13] "6264509100/6264509100_R03C01_Red.idat"     
[14] "6264509100/6264509100_R03C02_Grn.idat"     
[15] "6264509100/6264509100_R03C02_Red.idat"     
[16] "6264509100/6264509100_R04C01_Grn.idat"     
[17] "6264509100/6264509100_R04C01_Red.idat"     
[18] "6264509100/6264509100_R04C02_Grn.idat"     
[19] "6264509100/6264509100_R04C02_Red.idat"     
[20] "6264509100/6264509100_R05C01_Grn.idat"     
[21] "6264509100/6264509100_R05C01_Red.idat"     
[22] "6264509100/6264509100_R05C02_Grn.idat"     
[23] "6264509100/6264509100_R05C02_Red.idat"     
[24] "6264509100/6264509100_R06C01_Grn.idat"     
[25] "6264509100/6264509100_R06C01_Red.idat"     
[26] "6264509100/6264509100_R06C02_Grn.idat"     
[27] "6264509100/6264509100_R06C02_Red.idat"     
[28] "ageData.RData"                             
[29] "human_c2_v5.rdata"                         
[30] "model-based-cpg-islands-hg19-chr17.txt"    
[31] "SampleSheet.csv"                           
[32] "wgEncodeRegDnaseClusteredV3chr17.bed"      </code></pre>
</div>
</div>
</section>
<section id="load-the-data" class="level2">
<h2 class="anchored" data-anchor-id="load-the-data">2.2 Load the Data</h2>
<p>This is where we load the example data and prepare it for our analysis.</p>
<ul>
<li><p>Many methylation specific packages &nbsp;</p></li>
<li><p><code>Minfi</code> and <code>read.metharray.sheet</code></p></li>
</ul>
<div class="cell">

</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the 450k annotation data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>ann450k <span class="ot">&lt;-</span> <span class="fu">getAnnotation</span>(IlluminaHumanMethylation450kanno.ilmn12.hg19)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ann450k)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 6 rows and 33 columns
                   chr       pos      strand        Name    AddressA
           &lt;character&gt; &lt;integer&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
cg00050873        chrY   9363356           -  cg00050873    32735311
cg00212031        chrY  21239348           -  cg00212031    29674443
cg00213748        chrY   8148233           -  cg00213748    30703409
cg00214611        chrY  15815688           -  cg00214611    69792329
cg00455876        chrY   9385539           -  cg00455876    27653438
cg01707559        chrY   6778695           +  cg01707559    45652402
              AddressB              ProbeSeqA              ProbeSeqB
           &lt;character&gt;            &lt;character&gt;            &lt;character&gt;
cg00050873    31717405 ACAAAAAAACAACACACAAC.. ACGAAAAAACAACGCACAAC..
cg00212031    38703326 CCCAATTAACCACAAAAACT.. CCCAATTAACCGCAAAAACT..
cg00213748    36767301 TTTTAACACCTAACACCATT.. TTTTAACGCCTAACACCGTT..
cg00214611    46723459 CTAACTTCCAAACCACACTT.. CTAACTTCCGAACCGCGCTT..
cg00455876    69732350 AACTCTAAACTACCCAACAC.. AACTCTAAACTACCCGACAC..
cg01707559    64689504 ACAAATTAAAAACACTAAAA.. GCGAATTAAAAACACTAAAA..
                  Type    NextBase       Color    Probe_rs Probe_maf
           &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt;
cg00050873           I           A         Red          NA        NA
cg00212031           I           T         Red          NA        NA
cg00213748           I           A         Red          NA        NA
cg00214611           I           A         Red          NA        NA
cg00455876           I           A         Red          NA        NA
cg01707559           I           A         Red          NA        NA
                CpG_rs   CpG_maf      SBE_rs   SBE_maf           Islands_Name
           &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt;            &lt;character&gt;
cg00050873          NA        NA          NA        NA   chrY:9363680-9363943
cg00212031          NA        NA          NA        NA chrY:21238448-21240005
cg00213748          NA        NA          NA        NA   chrY:8147877-8148210
cg00214611          NA        NA          NA        NA chrY:15815488-15815779
cg00455876          NA        NA          NA        NA   chrY:9385471-9385777
cg01707559          NA        NA          NA        NA   chrY:6778574-6780028
           Relation_to_Island       Forward_Sequence              SourceSeq
                  &lt;character&gt;            &lt;character&gt;            &lt;character&gt;
cg00050873            N_Shore TATCTCTGTCTGGCGAGGAG.. CGGGGTCCACCCACTCCAAA..
cg00212031             Island CCATTGGCCCGCCCCAGTTG.. CGCACGTCTTCCCGACCGCA..
cg00213748            S_Shore TCTGTGGGACCATTTTAACG.. CGCCCCCTCCTGCAGAACCT..
cg00214611             Island GCGCCGGCAGGACTAGCTTC.. CGCCCGCGCCACACTGCAGC..
cg00455876             Island CGCGTGTGCCTGGACTCTGA.. GACTCTGAGCTACCCGGCAC..
cg01707559             Island AGCGGCCGCTCCCAGTGGTG.. CGCCCTCTGTCGCTGCAGCC..
           Random_Loci Methyl27_Loci UCSC_RefGene_Name UCSC_RefGene_Accession
           &lt;character&gt;   &lt;character&gt;       &lt;character&gt;            &lt;character&gt;
cg00050873                              TSPY4;FAM197Y2 NM_001164471;NR_001553
cg00212031                                      TTTY14              NR_001543
cg00213748                                                                   
cg00214611                               TMSB4Y;TMSB4Y    NM_004202;NM_004202
cg00455876                                                                   
cg01707559                           TBL1Y;TBL1Y;TBL1Y NM_134259;NM_033284;..
             UCSC_RefGene_Group     Phantom         DMR    Enhancer
                    &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
cg00050873         Body;TSS1500                                    
cg00212031               TSS200                                    
cg00213748                                                         
cg00214611        1stExon;5'UTR                                    
cg00455876                                                         
cg01707559 TSS200;TSS200;TSS200                                    
                    HMM_Island Regulatory_Feature_Name Regulatory_Feature_Group
                   &lt;character&gt;             &lt;character&gt;              &lt;character&gt;
cg00050873   Y:9973136-9976273                                                 
cg00212031 Y:19697854-19699393                                                 
cg00213748   Y:8207555-8208234                                                 
cg00214611 Y:14324883-14325218     Y:15815422-15815706   Promoter_Associated_..
cg00455876   Y:9993394-9995882                                                 
cg01707559   Y:6838022-6839951                                                 
                   DHS
           &lt;character&gt;
cg00050873            
cg00212031            
cg00213748            
cg00214611            
cg00455876            
cg01707559            </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the sample sheet for the experiment</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>targets <span class="ot">&lt;-</span> <span class="fu">read.metharray.sheet</span>(dataDirectory, <span class="at">pattern=</span><span class="st">"SampleSheet.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>[read.metharray.sheet] Found the following CSV files:</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/SampleSheet.csv"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>targets</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Sample_Name Sample_Well   Sample_Source Sample_Group Sample_Label Pool_ID
1            1          A1             M28        naive        naive    &lt;NA&gt;
2            2          B1             M28        rTreg        rTreg    &lt;NA&gt;
3            3          C1             M28    act_naive    act_naive    &lt;NA&gt;
4            4          D1             M29        naive        naive    &lt;NA&gt;
5            5          E1             M29    act_naive    act_naive    &lt;NA&gt;
6            6          F1             M29    act_rTreg    act_rTreg    &lt;NA&gt;
7            7          G1             M30        naive        naive    &lt;NA&gt;
8            8          H1             M30        rTreg        rTreg    &lt;NA&gt;
9            9          A2             M30    act_naive    act_naive    &lt;NA&gt;
10          10          B2             M30    act_rTreg    act_rTreg    &lt;NA&gt;
11          11         H06 VICS-72098-18-B        birth        birth    &lt;NA&gt;
    Array      Slide
1  R01C01 6264509100
2  R02C01 6264509100
3  R03C01 6264509100
4  R04C01 6264509100
5  R05C01 6264509100
6  R06C01 6264509100
7  R01C02 6264509100
8  R02C02 6264509100
9  R03C02 6264509100
10 R04C02 6264509100
11 R06C02 5975827018
                                                                                                       Basename
1  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R01C01
2  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R02C01
3  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R03C01
4  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R04C01
5  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R05C01
6  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R06C01
7  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R01C02
8  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R02C02
9  C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R03C02
10 C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/6264509100/6264509100_R04C02
11 C:/Users/legoj/AppData/Local/R/win-library/4.2/methylationArrayAnalysis/extdata/5975827018/5975827018_R06C02</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the raw data from the IDAT files</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>rgSet <span class="ot">&lt;-</span> <span class="fu">read.metharray.exp</span>(<span class="at">targets=</span>targets)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>rgSet</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RGChannelSet 
dim: 622399 11 
metadata(0):
assays(2): Green Red
rownames(622399): 10600313 10600322 ... 74810490 74810492
rowData names(0):
colnames(11): 6264509100_R01C01 6264509100_R02C01 ... 6264509100_R04C02
  5975827018_R06C02
colData names(10): Sample_Name Sample_Well ... Basename filenames
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12.hg19</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Give the samples descriptive names</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>targets<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">paste</span>(targets<span class="sc">$</span>Sample_Group,targets<span class="sc">$</span>Sample_Name,<span class="at">sep=</span><span class="st">"."</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sampleNames</span>(rgSet) <span class="ot">&lt;-</span> targets<span class="sc">$</span>ID</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>rgSet</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RGChannelSet 
dim: 622399 11 
metadata(0):
assays(2): Green Red
rownames(622399): 10600313 10600322 ... 74810490 74810492
rowData names(0):
colnames(11): naive.1 rTreg.2 ... act_rTreg.10 birth.11
colData names(10): Sample_Name Sample_Well ... Basename filenames
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12.hg19</code></pre>
</div>
</div>
</section>
<section id="quality-control" class="level2">
<h2 class="anchored" data-anchor-id="quality-control">2.3 Quality Control</h2>
<p>This is where we gauge the quality of our data and remove parts that aren’t good enough to use.</p>
<ul>
<li><p>P-value detection through <code>detP</code> function.&nbsp;</p></li>
<li><p>Plot the mean</p></li>
<li><p>QcReport function in <code>minfi</code></p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the detection p-values</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>detP <span class="ot">&lt;-</span> <span class="fu">detectionP</span>(rgSet)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(detP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 naive.1       rTreg.2  act_naive.3     naive.4   act_naive.5
cg00050873  0.000000e+00  0.000000e+00 0.000000e+00 0.00000e+00  0.000000e+00
cg00212031  0.000000e+00  0.000000e+00 0.000000e+00 0.00000e+00  0.000000e+00
cg00213748  2.139652e-88  4.213813e-31 1.181802e-12 1.29802e-47  8.255482e-15
cg00214611  0.000000e+00  0.000000e+00 0.000000e+00 0.00000e+00  0.000000e+00
cg00455876 1.400696e-234 9.349236e-111 4.272105e-90 0.00000e+00 3.347145e-268
cg01707559  0.000000e+00  0.000000e+00 0.000000e+00 0.00000e+00  0.000000e+00
             act_rTreg.6      naive.7       rTreg.8  act_naive.9  act_rTreg.10
cg00050873  0.000000e+00  0.00000e+00  0.000000e+00 0.000000e+00  0.000000e+00
cg00212031  0.000000e+00  0.00000e+00  0.000000e+00 0.000000e+00  0.000000e+00
cg00213748  2.592206e-23  1.16160e-28  1.469801e-05 1.543654e-21  1.365951e-08
cg00214611  0.000000e+00  0.00000e+00  0.000000e+00 0.000000e+00  0.000000e+00
cg00455876 4.690740e-308 1.08647e-219 5.362780e-178 0.000000e+00 7.950724e-295
cg01707559  0.000000e+00  0.00000e+00  0.000000e+00 0.000000e+00  0.000000e+00
                birth.11
cg00050873  0.000000e+00
cg00212031 2.638199e-237
cg00213748  6.735224e-01
cg00214611  7.344451e-01
cg00455876 4.403634e-174
cg01707559  0.000000e+00</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine mean detection p-values across all samples to identify any failed samples</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">"Dark2"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">colMeans</span>(detP), <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)], <span class="at">las=</span><span class="dv">2</span>, </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex.names=</span><span class="fl">0.8</span>, <span class="at">ylab=</span><span class="st">"Mean detection p-values"</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.05</span>,<span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">fill=</span>pal,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">colMeans</span>(detP), <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)], <span class="at">las=</span><span class="dv">2</span>, </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex.names=</span><span class="fl">0.8</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.002</span>), <span class="at">ylab=</span><span class="st">"Mean detection p-values"</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.05</span>,<span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">fill=</span>pal, </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">bg=</span><span class="st">"white"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE

Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by the
caller; using TRUE</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove poor quality samples</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">colMeans</span>(detP) <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>rgSet <span class="ot">&lt;-</span> rgSet[,keep]</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>rgSet</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RGChannelSet 
dim: 622399 10 
metadata(0):
assays(2): Green Red
rownames(622399): 10600313 10600322 ... 74810490 74810492
rowData names(0):
colnames(10): naive.1 rTreg.2 ... act_naive.9 act_rTreg.10
colData names(10): Sample_Name Sample_Well ... Basename filenames
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12.hg19</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove poor quality samples from targets data</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>targets <span class="ot">&lt;-</span> targets[keep,]</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>targets[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Sample_Name Sample_Well Sample_Source Sample_Group Sample_Label
1            1          A1           M28        naive        naive
2            2          B1           M28        rTreg        rTreg
3            3          C1           M28    act_naive    act_naive
4            4          D1           M29        naive        naive
5            5          E1           M29    act_naive    act_naive
6            6          F1           M29    act_rTreg    act_rTreg
7            7          G1           M30        naive        naive
8            8          H1           M30        rTreg        rTreg
9            9          A2           M30    act_naive    act_naive
10          10          B2           M30    act_rTreg    act_rTreg</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove poor quality samples from detection p-value table</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>detP <span class="ot">&lt;-</span> detP[,keep]</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(detP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 485512     10</code></pre>
</div>
</div>
</section>
<section id="normalization" class="level2">
<h2 class="anchored" data-anchor-id="normalization">2.4 Normalization</h2>
<p>Another process of data treatment to help eliminate unwanted variability between samples.</p>
<ul>
<li><p>Dealing with variation within data</p></li>
<li><p>Many different methods even within just <code>minfi</code></p></li>
<li><p><code>preprocessFunnorm</code> for global methylation differences</p></li>
<li><p>Quantile is more suited samples with no global differences</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize the data; this results in a GenomicRatioSet object</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>mSetSq <span class="ot">&lt;-</span> <span class="fu">preprocessQuantile</span>(rgSet) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>[preprocessQuantile] Mapping to genome.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .getSex(CN = CN, xIndex = xIndex, yIndex = yIndex, cutoff = cutoff):
An inconsistency was encountered while determining sex. One possibility is
that only one sex is present. We recommend further checks, for example with the
plotSex function.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[preprocessQuantile] Fixing outliers.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[preprocessQuantile] Quantile normalizing.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a MethylSet object from the raw data for plotting</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>mSetRaw <span class="ot">&lt;-</span> <span class="fu">preprocessRaw</span>(rgSet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise what the data looks like before and after normalization</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">densityPlot</span>(rgSet, <span class="at">sampGroups=</span>targets<span class="sc">$</span>Sample_Group,<span class="at">main=</span><span class="st">"Raw"</span>, <span class="at">legend=</span><span class="cn">FALSE</span>)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"top"</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), </span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">text.col=</span><span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">"Dark2"</span>))</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">densityPlot</span>(<span class="fu">getBeta</span>(mSetSq), <span class="at">sampGroups=</span>targets<span class="sc">$</span>Sample_Group,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">main=</span><span class="st">"Normalized"</span>, <span class="at">legend=</span><span class="cn">FALSE</span>)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"top"</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), </span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">text.col=</span><span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">"Dark2"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="data-exploration" class="level2">
<h2 class="anchored" data-anchor-id="data-exploration">2.5 Data Exploration</h2>
<p>Visualizing data is a key component of data exploration and can make it easier to understand results. In this workflow, we used multi-dimensional scaling plots (MDS). MDS plots are based on principal components analysis (PCA) and look at the similarities and differences between the various samples. Samples that are more like each other should cluster together, and samples that are different should be further apart on the plot.</p>
<p>Note: PCA is a data reduction method that reduces the number of variables that account for the variance in the data and makes it easier to interpret large datasets.</p>
<p>To plot this we used the <code>plotMDS</code> function, a built-in function in R to handle this kind of data. It appears like ggplot2 with the ability to alter specific aspects of the plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MDS plots to look at largest sources of variation</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSq), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)])</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"top"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">text.col=</span>pal,</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">bg=</span><span class="st">"white"</span>, <span class="at">cex=</span><span class="fl">0.7</span>)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSq), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>,  </span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)])</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"top"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)), <span class="at">text.col=</span>pal,</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">bg=</span><span class="st">"white"</span>, <span class="at">cex=</span><span class="fl">0.7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine higher dimensions to look at other sources of variation</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSq), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)], <span class="at">dim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"top"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">text.col=</span>pal, </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSq), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)], <span class="at">dim=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">text.col=</span>pal,</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSq), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)], <span class="at">dim=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>))</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">text.col=</span>pal,</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="filtering" class="level2">
<h2 class="anchored" data-anchor-id="filtering">2.6 Filtering</h2>
<p>Like what we did with Trimmomatic when working with databases, this step requires that we filter out DNA probes that interfere with our analysis. This is because the signal from these probes is unreliable and removing allows us to perform fewer statistical tests.</p>
<p>In this step, we begin by ordering the probes, then removing any probes that have failed in one or more samples.</p>
<p>While this dataset is entirely composed of male donors, filtering probes from both X and Y chromosomes can be done.</p>
<p>Similarly, single nucleotide polymorphisms (SNPs) - the most common type of genetic variation among people – can also be filtered out of probes using the function <code>minifi</code>.</p>
<p>After filtering we can re-examined the data using MDS plots to see if the relationship between samples changed. In this workflow, you can see that removing the SNP-affected CpGs probes from the data changes the sample clustering in the MDS plots.</p>
<p>We then calculated the M-values, a measure of the number of molecules that are methylated or unmethylated and the beta value which roughly represents the percentage of a site that is methylated (Du et al, 2010).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure probes are in the same order in the mSetSq and detP objects</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>detP <span class="ot">&lt;-</span> detP[<span class="fu">match</span>(<span class="fu">featureNames</span>(mSetSq),<span class="fu">rownames</span>(detP)),] </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any probes that have failed in one or more samples</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(detP <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">==</span> <span class="fu">ncol</span>(mSetSq) </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>keep
 FALSE   TRUE 
   977 484535 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>mSetSqFlt <span class="ot">&lt;-</span> mSetSq[keep,]</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>mSetSqFlt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: GenomicRatioSet 
dim: 484535 10 
metadata(0):
assays(2): M CN
rownames(484535): cg13869341 cg14008030 ... cg08265308 cg14273923
rowData names(0):
colnames(10): naive.1 rTreg.2 ... act_naive.9 act_rTreg.10
colData names(13): Sample_Name Sample_Well ... yMed predictedSex
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12.hg19
Preprocessing
  Method: Raw (no normalization or bg correction)
  minfi version: 1.44.0
  Manifest version: 0.4.0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># If your data includes males and females, remove probes on the sex chromosomes</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="sc">!</span>(<span class="fu">featureNames</span>(mSetSqFlt) <span class="sc">%in%</span> ann450k<span class="sc">$</span>Name[ann450k<span class="sc">$</span>chr <span class="sc">%in%</span> </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">c</span>(<span class="st">"chrX"</span>,<span class="st">"chrY"</span>)])</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>keep
 FALSE   TRUE 
 11608 472927 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>mSetSqFlt <span class="ot">&lt;-</span> mSetSqFlt[keep,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove probes with SNPs at CpG site</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>mSetSqFlt <span class="ot">&lt;-</span> <span class="fu">dropLociWithSnps</span>(mSetSqFlt)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>mSetSqFlt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: GenomicRatioSet 
dim: 455924 10 
metadata(0):
assays(2): M CN
rownames(455924): cg13869341 cg14008030 ... cg07660283 cg09226288
rowData names(0):
colnames(10): naive.1 rTreg.2 ... act_naive.9 act_rTreg.10
colData names(13): Sample_Name Sample_Well ... yMed predictedSex
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12.hg19
Preprocessing
  Method: Raw (no normalization or bg correction)
  minfi version: 1.44.0
  Manifest version: 0.4.0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Exclude cross reactive probes </span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>xReactiveProbes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file=</span><span class="fu">paste</span>(dataDirectory,</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"48639-non-specific-probes-Illumina450k.csv"</span>,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep=</span><span class="st">"/"</span>), <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="sc">!</span>(<span class="fu">featureNames</span>(mSetSqFlt) <span class="sc">%in%</span> xReactiveProbes<span class="sc">$</span>TargetID)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>keep
 FALSE   TRUE 
 26527 429397 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>mSetSqFlt <span class="ot">&lt;-</span> mSetSqFlt[keep,] </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>mSetSqFlt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: GenomicRatioSet 
dim: 429397 10 
metadata(0):
assays(2): M CN
rownames(429397): cg13869341 cg24669183 ... cg19565306 cg09226288
rowData names(0):
colnames(10): naive.1 rTreg.2 ... act_naive.9 act_rTreg.10
colData names(13): Sample_Name Sample_Well ... yMed predictedSex
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12.hg19
Preprocessing
  Method: Raw (no normalization or bg correction)
  minfi version: 1.44.0
  Manifest version: 0.4.0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSqFlt), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)], <span class="at">cex=</span><span class="fl">0.8</span>)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"right"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), <span class="at">text.col=</span>pal,</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.65</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSqFlt), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)])</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"right"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)), <span class="at">text.col=</span>pal,</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine higher dimensions to look at other sources of variation</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSqFlt), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)], <span class="at">dim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"right"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)), <span class="at">text.col=</span>pal,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSqFlt), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)], <span class="at">dim=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)), <span class="at">text.col=</span>pal,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">getM</span>(mSetSqFlt), <span class="at">top=</span><span class="dv">1000</span>, <span class="at">gene.selection=</span><span class="st">"common"</span>, </span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span>pal[<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)], <span class="at">dim=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>))</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"right"</span>, <span class="at">legend=</span><span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source)), <span class="at">text.col=</span>pal,</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">bg=</span><span class="st">"white"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate M-values for statistical analysis</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>mVals <span class="ot">&lt;-</span> <span class="fu">getM</span>(mSetSqFlt)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mVals[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             naive.1   rTreg.2 act_naive.3   naive.4 act_naive.5
cg13869341  2.421276  2.515948    2.165745  2.286314    2.109441
cg24669183  2.169414  2.235964    2.280734  1.632309    2.184435
cg15560884  1.761176  1.577578    1.597503  1.777486    1.764999
cg01014490 -3.504268 -3.825119   -5.384735 -4.537864   -4.296526
cg17505339  3.082191  3.924931    4.163206  3.255373    3.654134
cg11954957  1.546401  1.912204    1.727910  2.441267    1.618331</code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>bVals <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(mSetSqFlt)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bVals[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              naive.1    rTreg.2 act_naive.3    naive.4 act_naive.5
cg13869341 0.84267937 0.85118462   0.8177504 0.82987650  0.81186174
cg24669183 0.81812908 0.82489238   0.8293297 0.75610281  0.81967323
cg15560884 0.77219626 0.74903910   0.7516263 0.77417882  0.77266205
cg01014490 0.08098986 0.06590459   0.0233755 0.04127262  0.04842397
cg17505339 0.89439216 0.93822870   0.9471357 0.90520570  0.92641305
cg11954957 0.74495496 0.79008516   0.7681146 0.84450764  0.75431167</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">densityPlot</span>(bVals, <span class="at">sampGroups=</span>targets<span class="sc">$</span>Sample_Group, <span class="at">main=</span><span class="st">"Beta values"</span>, </span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend=</span><span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">"Beta values"</span>)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"top"</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), </span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">text.col=</span><span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">"Dark2"</span>))</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="fu">densityPlot</span>(mVals, <span class="at">sampGroups=</span>targets<span class="sc">$</span>Sample_Group, <span class="at">main=</span><span class="st">"M-values"</span>, </span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend=</span><span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">"M values"</span>)</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)), </span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">text.col=</span><span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">"Dark2"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="probe-wise-differential-methylation-analysis" class="level2">
<h2 class="anchored" data-anchor-id="probe-wise-differential-methylation-analysis">2.7 Probe-Wise Differential Methylation Analysis</h2>
<p>The focus of this dataset is discovering differentially methylated probes between different cell types. This style of analysis is called a paired analysis; differences between cell types are calculated within each individual, and then these differences are averaged across individuals to determine whether there is an overall significant difference in the mean methylation level for each CpG site.</p>
<p>We were interested in pairwise comparisons between the four cell types. In this workflow, we analyzed the M-values in limma. This allowed us to obtain moderated t-statistics and associated p-values for each CpG site, sites where a cytosine is followed by guanine.</p>
<p>Based on a false discovery rate of 5%, we found 3021 significantly differentially methylated CpGs in the naïve vs rTreg comparison. The rTreg/ act_rTreg did not show any significant differential methylation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the factor of interest</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>cellType <span class="ot">&lt;-</span> <span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the individual effect that we need to account for</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>individual <span class="ot">&lt;-</span> <span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Source) </span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the above to create a design matrix</span></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>cellType<span class="sc">+</span>individual, <span class="at">data=</span>targets)</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">levels</span>(cellType),<span class="fu">levels</span>(individual)[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the linear model </span></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(mVals, design)</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a><span class="co"># create a contrast matrix for specific comparisons</span></span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>contMatrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(naive<span class="sc">-</span>rTreg,</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>                           naive<span class="sc">-</span>act_naive,</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>                           rTreg<span class="sc">-</span>act_rTreg,</span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>                           act_naive<span class="sc">-</span>act_rTreg,</span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels=</span>design)</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>contMatrix</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           Contrasts
Levels      naive - rTreg naive - act_naive rTreg - act_rTreg
  act_naive             0                -1                 0
  act_rTreg             0                 0                -1
  naive                 1                 1                 0
  rTreg                -1                 0                 1
  M29                   0                 0                 0
  M30                   0                 0                 0
           Contrasts
Levels      act_naive - act_rTreg
  act_naive                     1
  act_rTreg                    -1
  naive                         0
  rTreg                         0
  M29                           0
  M30                           0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the contrasts</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contMatrix)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2)</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a><span class="co"># look at the numbers of DM CpGs at FDR &lt; 0.05</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">decideTests</span>(fit2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       naive - rTreg naive - act_naive rTreg - act_rTreg act_naive - act_rTreg
Down            1606               408                 0                   552
NotSig        426397            428759            429397                427943
Up              1394               230                 0                   902</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the table of results for the first contrast (naive - rTreg)</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>ann450kSub <span class="ot">&lt;-</span> ann450k[<span class="fu">match</span>(<span class="fu">rownames</span>(mVals),ann450k<span class="sc">$</span>Name),</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">12</span><span class="sc">:</span><span class="dv">19</span>,<span class="dv">24</span><span class="sc">:</span><span class="fu">ncol</span>(ann450k))]</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>DMPs <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">num=</span><span class="cn">Inf</span>, <span class="at">coef=</span><span class="dv">1</span>, <span class="at">genelist=</span>ann450kSub)</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DMPs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             chr       pos strand       Name Probe_rs Probe_maf CpG_rs CpG_maf
cg07499259  chr1  12188502      + cg07499259     &lt;NA&gt;        NA   &lt;NA&gt;      NA
cg26992245  chr8  29848579      - cg26992245     &lt;NA&gt;        NA   &lt;NA&gt;      NA
cg09747445 chr15  70387268      - cg09747445     &lt;NA&gt;        NA   &lt;NA&gt;      NA
cg18808929  chr8  61825469      - cg18808929     &lt;NA&gt;        NA   &lt;NA&gt;      NA
cg25015733  chr2  99342986      - cg25015733     &lt;NA&gt;        NA   &lt;NA&gt;      NA
cg21179654  chr3 114057297      + cg21179654     &lt;NA&gt;        NA   &lt;NA&gt;      NA
           SBE_rs SBE_maf            Islands_Name Relation_to_Island
cg07499259   &lt;NA&gt;      NA                                    OpenSea
cg26992245   &lt;NA&gt;      NA                                    OpenSea
cg09747445   &lt;NA&gt;      NA chr15:70387929-70393206            N_Shore
cg18808929   &lt;NA&gt;      NA  chr8:61822358-61823028            S_Shelf
cg25015733   &lt;NA&gt;      NA  chr2:99346882-99348177            N_Shelf
cg21179654   &lt;NA&gt;      NA                                    OpenSea
                                          UCSC_RefGene_Name
cg07499259                                  TNFRSF8;TNFRSF8
cg26992245                                                 
cg09747445                                   TLE3;TLE3;TLE3
cg18808929                                                 
cg25015733                                           MGAT4A
cg21179654 ZBTB20;ZBTB20;ZBTB20;ZBTB20;ZBTB20;ZBTB20;ZBTB20
                                                                            UCSC_RefGene_Accession
cg07499259                                                                     NM_152942;NM_001243
cg26992245                                                                                        
cg09747445                                                        NM_001105192;NM_020908;NM_005078
cg18808929                                                                                        
cg25015733                                                                               NM_012214
cg21179654 NM_001164343;NM_001164346;NM_001164345;NM_001164342;NM_001164344;NM_001164347;NM_015642
                                  UCSC_RefGene_Group Phantom DMR Enhancer
cg07499259                                5'UTR;Body                     
cg26992245                                                           TRUE
cg09747445                            Body;Body;Body                     
cg18808929                                                           TRUE
cg25015733                                     5'UTR                     
cg21179654 3'UTR;3'UTR;3'UTR;3'UTR;3'UTR;3'UTR;3'UTR                     
                    HMM_Island Regulatory_Feature_Name
cg07499259 1:12111023-12111225                        
cg26992245                                            
cg09747445                                            
cg18808929                                            
cg25015733                                            
cg21179654                       3:114057192-114057775
                  Regulatory_Feature_Group DHS     logFC     AveExpr         t
cg07499259                                      3.654104  2.46652171  18.77533
cg26992245                                      4.450696 -0.09180715  18.35830
cg09747445                                     -3.337299 -0.25201484 -18.29169
cg18808929                                     -2.990263  0.77522878 -17.95543
cg25015733                                     -3.054336  0.83280190 -17.37376
cg21179654 Unclassified_Cell_type_specific      2.859016  1.32460816  17.33058
                P.Value   adj.P.Val        B
cg07499259 7.038346e-08 0.005176247 7.491458
cg26992245 8.386780e-08 0.005176247 7.394212
cg09747445 8.627940e-08 0.005176247 7.378282
cg18808929 9.970727e-08 0.005176247 7.296126
cg25015733 1.288536e-07 0.005176247 7.146883
cg21179654 1.313725e-07 0.005176247 7.135430</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(DMPs, <span class="at">file=</span><span class="st">"DMPs.csv"</span>, <span class="at">sep=</span><span class="st">","</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the top 4 most significantly differentially methylated CpGs </span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(<span class="fu">rownames</span>(DMPs)[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="cf">function</span>(cpg){</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCpg</span>(bVals, <span class="at">cpg=</span>cpg, <span class="at">pheno=</span>targets<span class="sc">$</span>Sample_Group, <span class="at">ylab =</span> <span class="st">"Beta values"</span>)</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$cg07499259
NULL

$cg26992245
NULL

$cg09747445
NULL

$cg18808929
NULL</code></pre>
</div>
</div>
</section>
<section id="differential-methylation-analysis-of-regions" class="level2">
<h2 class="anchored" data-anchor-id="differential-methylation-analysis-of-regions">2.8 Differential Methylation Analysis of Regions</h2>
<p>Our probe-wise differential methylation analysis can be valuable, but it is not alone sufficient if we want to look at entire regions. Several Bioconductor packages exist that can help obtain a broader overview of differential methylated regions (DMRs). This workflow utilizes the <code>dmrcate</code> Bioconductor package. The final figure maps the methylated regions against chromosome length and the known genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>myAnnotation <span class="ot">&lt;-</span> <span class="fu">cpg.annotate</span>(<span class="at">object =</span> mVals, <span class="at">datatype =</span> <span class="st">"array"</span>, <span class="at">what =</span> <span class="st">"M"</span>, </span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">analysis.type =</span> <span class="st">"differential"</span>, <span class="at">design =</span> design, </span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">contrasts =</span> <span class="cn">TRUE</span>, <span class="at">cont.matrix =</span> contMatrix, </span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">coef =</span> <span class="st">"naive - rTreg"</span>, <span class="at">arraytype =</span> <span class="st">"450K"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Your contrast returned 3000 individually significant probes. We recommend the default setting of pcutoff in dmrcate().</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(myAnnotation)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'CpGannotated' [package "DMRcate"] with 1 slot
  ..@ ranges:Formal class 'GRanges' [package "GenomicRanges"] with 7 slots
  .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots
  .. .. .. .. ..@ values         : Factor w/ 22 levels "chr1","chr2",..: 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. .. ..@ lengths        : int [1:22] 42733 31682 23086 18431 22093 32652 26437 18956 8961 22163 ...
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots
  .. .. .. .. ..@ start          : int [1:429397] 15865 534242 710097 714177 720865 758829 763119 779995 805102 805338 ...
  .. .. .. .. ..@ width          : int [1:429397] 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..@ NAMES          : chr [1:429397] "cg13869341" "cg24669183" "cg15560884" "cg01014490" ...
  .. .. .. .. ..@ elementType    : chr "ANY"
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots
  .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": 3
  .. .. .. .. ..@ lengths        : int 429397
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots
  .. .. .. .. ..@ seqnames   : chr [1:22] "chr1" "chr2" "chr3" "chr4" ...
  .. .. .. .. ..@ seqlengths : int [1:22] NA NA NA NA NA NA NA NA NA NA ...
  .. .. .. .. ..@ is_circular: logi [1:22] NA NA NA NA NA NA ...
  .. .. .. .. ..@ genome     : chr [1:22] NA NA NA NA ...
  .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots
  .. .. .. .. ..@ rownames       : NULL
  .. .. .. .. ..@ nrows          : int 429397
  .. .. .. .. ..@ elementType    : chr "ANY"
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. .. .. ..@ listData       :List of 4
  .. .. .. .. .. ..$ stat   : num [1:429397] 0.049 -2.0817 0.7734 -0.0304 -0.7651 ...
  .. .. .. .. .. ..$ diff   : num [1:429397] 0.00039 -0.04534 0.01594 0.00251 -0.00869 ...
  .. .. .. .. .. ..$ ind.fdr: num [1:429397] 0.994 0.562 0.871 0.997 0.873 ...
  .. .. .. .. .. ..$ is.sig : logi [1:429397] FALSE FALSE FALSE FALSE FALSE FALSE ...
  .. .. ..@ elementType    : chr "ANY"
  .. .. ..@ metadata       : list()</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co">#endif /* NEWSTUFF */</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>DMRs <span class="ot">&lt;-</span> <span class="fu">dmrcate</span>(myAnnotation, <span class="at">lambda=</span><span class="dv">1000</span>, <span class="at">C=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr1...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr2...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr3...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr4...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr5...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr6...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr7...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr8...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr9...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr10...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr11...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr12...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr13...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr14...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr15...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr16...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr17...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr18...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr19...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr20...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr21...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting chr22...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Demarcating regions...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Done!</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>results.ranges <span class="ot">&lt;-</span> <span class="fu">extractRanges</span>(DMRs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>snapshotDate(): 2022-10-31</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?DMRcatedata and browseVignettes('DMRcatedata') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>results.ranges</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 538 ranges and 8 metadata columns:
        seqnames              ranges strand |   no.cpgs min_smoothed_fdr
           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;        &lt;numeric&gt;
    [1]    chr17   57915665-57918682      * |        12      2.28354e-91
    [2]     chr3 114012316-114012912      * |         5     3.03677e-181
    [3]    chr18   21452730-21453131      * |         7     1.54955e-115
    [4]    chr17   74639731-74640078      * |         6      3.26482e-90
    [5]    chr15   29212889-29213860      * |         7      4.26302e-98
    ...      ...                 ...    ... .       ...              ...
  [534]     chr6   31832650-31833452      * |        18      1.66423e-28
  [535]     chr6   28830789-28831683      * |        14      8.41531e-63
  [536]     chr6 144385771-144387124      * |        22      1.30124e-60
  [537]     chr2   43454687-43455103      * |        15      8.51286e-26
  [538]     chr2   25141532-25142229      * |         8      2.72475e-25
           Stouffer      HMFDR      Fisher   maxdiff   meandiff
          &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
    [1] 5.40171e-10 0.02303084 5.50243e-08  0.398286   0.313161
    [2] 1.49513e-07 0.00711817 1.39012e-06  0.543428   0.425162
    [3] 6.77644e-07 0.01208830 1.63686e-06 -0.386747  -0.254609
    [4] 1.31264e-07 0.01359223 2.26745e-06 -0.252864  -0.195190
    [5] 7.19245e-06 0.01494937 4.06779e-06 -0.445320  -0.265725
    ...         ...        ...         ...       ...        ...
  [534]    0.881428  0.2270224    0.634763  0.153367  0.0490080
  [535]    0.761269  0.0733253    0.644056 -0.248661 -0.0338383
  [536]    0.996402  0.0775623    0.678935  0.325422  0.0449451
  [537]    0.974107  0.1582613    0.680381 -0.218836 -0.0413772
  [538]    0.992175  0.0552720    0.743929  0.282058  0.0314244
        overlapping.genes
              &lt;character&gt;
    [1]       VMP1, MIR21
    [2]             TIGIT
    [3]             LAMA3
    [4]        ST6GALNAC1
    [5]             APBA2
    ...               ...
  [534]           SLC44A4
  [535]  XXbac-BPG308K3.6
  [536]              &lt;NA&gt;
  [537]             THADA
  [538]             ADCY3
  -------
  seqinfo: 22 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up the grouping variables and colous</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> pal[<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">unique</span>(targets<span class="sc">$</span>Sample_Group))]</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(groups) <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group))</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> groups[<span class="fu">as.character</span>(<span class="fu">factor</span>(targets<span class="sc">$</span>Sample_Group))]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw the plot for the top DMR</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DMR.plot</span>(<span class="at">ranges =</span> results.ranges, <span class="at">dmr =</span> <span class="dv">2</span>, <span class="at">CpGs =</span> bVals, <span class="at">phen.col =</span> cols, </span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">what =</span> <span class="st">"Beta"</span>, <span class="at">arraytype =</span> <span class="st">"450K"</span>, <span class="at">genome =</span> <span class="st">"hg19"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>snapshotDate(): 2022-10-31</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?DMRcatedata and browseVignettes('DMRcatedata') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in updateObjectFromSlots(object, ..., verbose = verbose): dropping
slot(s) 'columns' from object = 'GeneRegionTrack'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="customising-visualisations-of-methylation-data" class="level2">
<h2 class="anchored" data-anchor-id="customising-visualisations-of-methylation-data">2.9 Customising Visualisations of Methylation Data</h2>
<p>The <code>Gviz</code> package utilized by this workflow allows the customization of data visualization plots with a variety of different functions. Here we focus on the DMR identified in our previous graph, while also more clearly presenting the methylation data. There are other types of analysis that could be done with this workflow, but this covers the main functions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># indicate which genome is being used</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="st">"hg19"</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="co"># the index of the DMR that we will plot </span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>dmrIndex <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a><span class="co"># extract chromosome number and location from DMR results </span></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>chrom <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">seqnames</span>(results.ranges[dmrIndex]))</span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>start <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">start</span>(results.ranges[dmrIndex]))</span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>end <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">end</span>(results.ranges[dmrIndex]))</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a><span class="co"># add 25% extra space to plot</span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>minbase <span class="ot">&lt;-</span> start <span class="sc">-</span> (<span class="fl">0.25</span><span class="sc">*</span>(end<span class="sc">-</span>start))</span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>maxbase <span class="ot">&lt;-</span> end <span class="sc">+</span> (<span class="fl">0.25</span><span class="sc">*</span>(end<span class="sc">-</span>start))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># CpG islands</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>islandHMM <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(dataDirectory,</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"/model-based-cpg-islands-hg19-chr17.txt"</span>),</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>, <span class="at">header=</span><span class="cn">FALSE</span>)</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(islandHMM)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               V1     V2     V3   V4  V5   V6    V7    V8
1 chr17_ctg5_hap1   8935  10075 1141 129  815 0.714 0.887
2 chr17_ctg5_hap1  64252  64478  227  30  165 0.727 1.014
3 chr17_ctg5_hap1  87730  89480 1751 135 1194 0.682 0.663
4 chr17_ctg5_hap1  98265  98591  327  29  226 0.691 0.744
5 chr17_ctg5_hap1 120763 125451 4689 359 3032 0.647 0.733
6 chr17_ctg5_hap1 146257 146607  351  19  231 0.658 0.500</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>islandData <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="fu">Rle</span>(islandHMM[,<span class="dv">1</span>]), </span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ranges=</span><span class="fu">IRanges</span>(<span class="at">start=</span>islandHMM[,<span class="dv">2</span>], <span class="at">end=</span>islandHMM[,<span class="dv">3</span>]),</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">strand=</span><span class="fu">Rle</span>(<span class="fu">strand</span>(<span class="fu">rep</span>(<span class="st">"*"</span>,<span class="fu">nrow</span>(islandHMM)))))</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>islandData</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 3456 ranges and 0 metadata columns:
                seqnames            ranges strand
                   &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
     [1] chr17_ctg5_hap1        8935-10075      *
     [2] chr17_ctg5_hap1       64252-64478      *
     [3] chr17_ctg5_hap1       87730-89480      *
     [4] chr17_ctg5_hap1       98265-98591      *
     [5] chr17_ctg5_hap1     120763-125451      *
     ...             ...               ...    ...
  [3452]           chr17 81147380-81147511      *
  [3453]           chr17 81147844-81148321      *
  [3454]           chr17 81152612-81153665      *
  [3455]           chr17 81156194-81156512      *
  [3456]           chr17 81162945-81165532      *
  -------
  seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DNAseI hypersensitive sites</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>dnase <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(dataDirectory,<span class="st">"/wgEncodeRegDnaseClusteredV3chr17.bed"</span>),</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>,<span class="at">header=</span><span class="cn">FALSE</span>)</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dnase)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     V1   V2   V3 V4  V5 V6                                       V7
1 chr17  125  335  7 444  7                    84,83,88,90,77,87,89,
2 chr17  685  835  1 150  1                                      80,
3 chr17 2440 2675 13 410 13 0,30,102,104,38,47,61,68,122,1,51,73,75,
4 chr17 3020 3170  1 247  1                                     120,
5 chr17 3740 3890  2 161  2                                   71,73,
6 chr17 5520 6110  4 241  5                          17,19,25,16,16,
                                              V8
1                   328,208,444,218,109,171,191,
2                                           150,
3 204,410,301,206,46,48,84,164,85,12,98,215,146,
4                                           247,
5                                       108,161,
6                             241,185,239,26,52,</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>dnaseData <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span>dnase[,<span class="dv">1</span>],</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ranges=</span><span class="fu">IRanges</span>(<span class="at">start=</span>dnase[,<span class="dv">2</span>], <span class="at">end=</span>dnase[,<span class="dv">3</span>]),</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">strand=</span><span class="fu">Rle</span>(<span class="fu">rep</span>(<span class="st">"*"</span>,<span class="fu">nrow</span>(dnase))),</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data=</span>dnase[,<span class="dv">5</span>])</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>dnaseData</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 74282 ranges and 1 metadata column:
          seqnames            ranges strand |      data
             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
      [1]    chr17           125-335      * |       444
      [2]    chr17           685-835      * |       150
      [3]    chr17         2440-2675      * |       410
      [4]    chr17         3020-3170      * |       247
      [5]    chr17         3740-3890      * |       161
      ...      ...               ...    ... .       ...
  [74278]    chr17 81153140-81153350      * |       574
  [74279]    chr17 81153580-81153810      * |       208
  [74280]    chr17 81185540-81185750      * |       326
  [74281]    chr17 81188880-81189090      * |       209
  [74282]    chr17 81194900-81195115      * |       185
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>iTrack <span class="ot">&lt;-</span> <span class="fu">IdeogramTrack</span>(<span class="at">genome =</span> gen, <span class="at">chromosome =</span> chrom, <span class="at">name=</span><span class="st">""</span>)</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>gTrack <span class="ot">&lt;-</span> <span class="fu">GenomeAxisTrack</span>(<span class="at">col=</span><span class="st">"black"</span>, <span class="at">cex=</span><span class="dv">1</span>, <span class="at">name=</span><span class="st">""</span>, <span class="at">fontcolor=</span><span class="st">"black"</span>)</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>rTrack <span class="ot">&lt;-</span> <span class="fu">UcscTrack</span>(<span class="at">genome=</span>gen, <span class="at">chromosome=</span>chrom, <span class="at">track=</span><span class="st">"NCBI RefSeq"</span>, </span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">from=</span>minbase, <span class="at">to=</span>maxbase, <span class="at">trackType=</span><span class="st">"GeneRegionTrack"</span>, </span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rstarts=</span><span class="st">"exonStarts"</span>, <span class="at">rends=</span><span class="st">"exonEnds"</span>, <span class="at">gene=</span><span class="st">"name"</span>, </span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">symbol=</span><span class="st">"name2"</span>, <span class="at">transcript=</span><span class="st">"name"</span>, <span class="at">strand=</span><span class="st">"strand"</span>, </span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span><span class="st">"darkblue"</span>,<span class="at">stacking=</span><span class="st">"squish"</span>, <span class="at">name=</span><span class="st">"RefSeq"</span>, </span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">showId=</span><span class="cn">TRUE</span>, <span class="at">geneSymbol=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .local(x, ...): 'track' parameter is deprecated now you go by the 'table' instead
                Use ucscTables(genome, track) to retrieve the list of tables for a track

Warning in .local(x, ...): 'track' parameter is deprecated now you go by the 'table' instead
                Use ucscTables(genome, track) to retrieve the list of tables for a track</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>ann450kOrd <span class="ot">&lt;-</span> ann450kSub[<span class="fu">order</span>(ann450kSub<span class="sc">$</span>chr,ann450kSub<span class="sc">$</span>pos),]</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ann450kOrd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 6 rows and 22 columns
                   chr       pos      strand        Name    Probe_rs Probe_maf
           &lt;character&gt; &lt;integer&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt;
cg13869341        chr1     15865           +  cg13869341          NA        NA
cg24669183        chr1    534242           -  cg24669183   rs6680725  0.108100
cg15560884        chr1    710097           +  cg15560884          NA        NA
cg01014490        chr1    714177           -  cg01014490          NA        NA
cg17505339        chr1    720865           -  cg17505339          NA        NA
cg11954957        chr1    758829           +  cg11954957 rs115498424  0.029514
                CpG_rs   CpG_maf      SBE_rs   SBE_maf       Islands_Name
           &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt;        &lt;character&gt;
cg13869341          NA        NA          NA        NA                   
cg24669183          NA        NA          NA        NA chr1:533219-534114
cg15560884          NA        NA          NA        NA chr1:713984-714547
cg01014490          NA        NA          NA        NA chr1:713984-714547
cg17505339          NA        NA          NA        NA                   
cg11954957          NA        NA          NA        NA chr1:762416-763445
           Relation_to_Island UCSC_RefGene_Name UCSC_RefGene_Accession
                  &lt;character&gt;       &lt;character&gt;            &lt;character&gt;
cg13869341            OpenSea            WASH5P              NR_024540
cg24669183            S_Shore                                         
cg15560884            N_Shelf                                         
cg01014490             Island                                         
cg17505339            OpenSea                                         
cg11954957            N_Shelf                                         
           UCSC_RefGene_Group     Phantom         DMR    Enhancer
                  &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
cg13869341               Body                                    
cg24669183                                                       
cg15560884                                                       
cg01014490                                                       
cg17505339                                                       
cg11954957                                                       
                HMM_Island Regulatory_Feature_Name Regulatory_Feature_Group
               &lt;character&gt;             &lt;character&gt;              &lt;character&gt;
cg13869341                                                                 
cg24669183 1:523025-524193                                                 
cg15560884                                                                 
cg01014490 1:703784-704410         1:713802-715219      Promoter_Associated
cg17505339                                                                 
cg11954957                                                                 
                   DHS
           &lt;character&gt;
cg13869341            
cg24669183            
cg15560884            
cg01014490            
cg17505339            
cg11954957            </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>bValsOrd <span class="ot">&lt;-</span> bVals[<span class="fu">match</span>(ann450kOrd<span class="sc">$</span>Name,<span class="fu">rownames</span>(bVals)),]</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bValsOrd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              naive.1    rTreg.2 act_naive.3    naive.4 act_naive.5 act_rTreg.6
cg13869341 0.84267937 0.85118462   0.8177504 0.82987650  0.81186174   0.8090798
cg24669183 0.81812908 0.82489238   0.8293297 0.75610281  0.81967323   0.8187838
cg15560884 0.77219626 0.74903910   0.7516263 0.77417882  0.77266205   0.7721528
cg01014490 0.08098986 0.06590459   0.0233755 0.04127262  0.04842397   0.0644404
cg17505339 0.89439216 0.93822870   0.9471357 0.90520570  0.92641305   0.9286016
cg11954957 0.74495496 0.79008516   0.7681146 0.84450764  0.75431167   0.8116911
             naive.7    rTreg.8 act_naive.9 act_rTreg.10
cg13869341 0.8891851 0.88537940  0.90916748   0.88334231
cg24669183 0.7903763 0.85304116  0.80930568   0.80979554
cg15560884 0.7658623 0.75909061  0.78099397   0.78569274
cg01014490 0.0245281 0.02832358  0.07740468   0.04640659
cg17505339 0.8889361 0.87205348  0.90099782   0.93508348
cg11954957 0.7832207 0.84929777  0.84719430   0.83350220</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create genomic ranges object from methylation data</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>cpgData <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="fu">Rle</span>(ann450kOrd<span class="sc">$</span>chr),</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ranges=</span><span class="fu">IRanges</span>(<span class="at">start=</span>ann450kOrd<span class="sc">$</span>pos, <span class="at">end=</span>ann450kOrd<span class="sc">$</span>pos),</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">strand=</span><span class="fu">Rle</span>(<span class="fu">rep</span>(<span class="st">"*"</span>,<span class="fu">nrow</span>(ann450kOrd))),</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">betas=</span>bValsOrd)</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a><span class="co"># extract data on CpGs in DMR</span></span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>cpgData <span class="ot">&lt;-</span> <span class="fu">subsetByOverlaps</span>(cpgData, results.ranges[dmrIndex])</span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a><span class="co"># methylation data track</span></span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>methTrack <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="at">range=</span>cpgData, <span class="at">groups=</span>targets<span class="sc">$</span>Sample_Group,<span class="at">genome =</span> gen,</span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">chromosome=</span>chrom, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.05</span>,<span class="fl">1.05</span>), <span class="at">col=</span>pal,</span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">type=</span><span class="fu">c</span>(<span class="st">"a"</span>,<span class="st">"p"</span>), <span class="at">name=</span><span class="st">"DNA Meth.</span><span class="sc">\n</span><span class="st">(beta value)"</span>,</span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">background.panel=</span><span class="st">"white"</span>, <span class="at">legend=</span><span class="cn">TRUE</span>, <span class="at">cex.title=</span><span class="fl">0.8</span>,</span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">cex.axis=</span><span class="fl">0.8</span>, <span class="at">cex.legend=</span><span class="fl">0.8</span>)</span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a><span class="co"># CpG island track</span></span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a>islandTrack <span class="ot">&lt;-</span> <span class="fu">AnnotationTrack</span>(<span class="at">range=</span>islandData, <span class="at">genome=</span>gen, <span class="at">name=</span><span class="st">"CpG Is."</span>, </span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>                               <span class="at">chromosome=</span>chrom,<span class="at">fill=</span><span class="st">"darkgreen"</span>)</span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a><span class="co"># DNaseI hypersensitive site data track</span></span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a>dnaseTrack <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="at">range=</span>dnaseData, <span class="at">genome=</span>gen, <span class="at">name=</span><span class="st">"DNAseI"</span>, </span>
<span id="cb119-21"><a href="#cb119-21" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type=</span><span class="st">"gradient"</span>, <span class="at">chromosome=</span>chrom)</span>
<span id="cb119-22"><a href="#cb119-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-23"><a href="#cb119-23" aria-hidden="true" tabindex="-1"></a><span class="co"># DMR position data track</span></span>
<span id="cb119-24"><a href="#cb119-24" aria-hidden="true" tabindex="-1"></a>dmrTrack <span class="ot">&lt;-</span> <span class="fu">AnnotationTrack</span>(<span class="at">start=</span>start, <span class="at">end=</span>end, <span class="at">genome=</span>gen, <span class="at">name=</span><span class="st">"DMR"</span>, </span>
<span id="cb119-25"><a href="#cb119-25" aria-hidden="true" tabindex="-1"></a>                            <span class="at">chromosome=</span>chrom,<span class="at">fill=</span><span class="st">"darkred"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>tracks <span class="ot">&lt;-</span> <span class="fu">list</span>(iTrack, gTrack, methTrack, dmrTrack, islandTrack, dnaseTrack,</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>               rTrack)</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>sizes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>) <span class="co"># set up the relative sizes of the tracks</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTracks</span>(tracks, <span class="at">from=</span>minbase, <span class="at">to=</span>maxbase, <span class="at">showTitle=</span><span class="cn">TRUE</span>, <span class="at">add53=</span><span class="cn">TRUE</span>, </span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">add35=</span><span class="cn">TRUE</span>, <span class="at">grid=</span><span class="cn">TRUE</span>, <span class="at">lty.grid=</span><span class="dv">3</span>, <span class="at">sizes =</span> sizes, <span class="fu">length</span>(tracks))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="BioshockDoc_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="conclusions" class="level2">
<h2 class="anchored" data-anchor-id="conclusions">3.1 Conclusions</h2>
<p><strong>Things Learned</strong></p>
<ul>
<li><p>How to navigate the Bioconductor website</p></li>
<li><p>Using Bioconductor for supplemental research</p></li>
<li><p>R has many packages to handle a wide variety of biological data</p></li>
<li><p>How to use Bioconductor to learn how to approach different data problems</p></li>
<li><p>Methylation is complicated</p></li>
</ul>
<p><strong>Challenges</strong></p>
<ul>
<li><p>Undoing commits with GitKraken</p></li>
<li><p>Updating R and not just RStudio</p></li>
<li><p>Getting the document to render</p></li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>